verifier/README.md

```diff
@@ -6,18 +6,30 @@ While [Miden](../miden) crate also contains verifier functionality, if a project
 ## Usage
 This crate exposes a `verify()` function which can be used to verify proofs of program execution. The function takes the following parameters:
 
-* `program_hash: &[u8; 32]` - an array of 32 bytes representing a hash of the program to be verified.
-* `public_inputs: &[u128]` - a list of public inputs against which the program was executed.
-* `outputs: &[u128]` - a list of outputs generated by the program.
-* `proof: &StarkProof` - the proof generated during program execution.
+* `program_hash: Digest` - a hash of the program to be verified (represented as a 32-byte digest).
+* `stack_inputs: &[u64]` - a list of values with the stack was initialized prior to a program's execution.
+* `stack_outputs: &[u64]` - a list of values returned from the stack after a program's execution completes.
+* `proof: StarkProof` - the proof generated during program execution.
 
-The function returns `Result<(), VerifierError>` which will be `Ok(())` if verification passes, or `Err(VerifierError)` if verification fails, with `VerifierError` describing the reason for the failure.
+Stack inputs are expected to be ordered as if they would be pushed onto the stack one by one. Thus, their expected order on the stack will be the reverse of the order in which they are provided, and the last value in the `stack_inputs` slice is expected to be the value at the top of the stack.
+
+Stack outputs are expected to be ordered as if they would be popped off the stack one by one. Thus, the value at the top of the stack is expected to be in the first position of the `stack_outputs` slice, and the order of the rest of the output elements will also match the order on the stack. This the reverse of order of `stack_inputs` slice.
```

Comment:
nit: the last sentence has the same problem here as I noted earlier, so probably it needs to be updated wherever the original source is (somewhere in the docs?).

"This is the reverse of the order of the `stack_inputs` slice."